### PR TITLE
Allow field and facet presenters to be configured

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/helpers/blacklight/facets_helper_behavior.rb [hunk 1]
```diff
@@ -169,7 +169,8 @@ def facet_partial_name(display_facet = nil)
   deprecation_deprecate :facet_partial_name
 
   def facet_field_presenter(facet_config, display_facet)
-    Blacklight::FacetFieldPresenter.new(facet_config, display_facet, self)
+    presenter_class = facet_config.presenter || Blacklight::FacetFieldPresenter
```
Comment (jcoyne) [inline app/helpers/blacklight/facets_helper_behavior.rb:172]: could we set this default in the facet_config itself to avoid the conditional here?

Comment (cbeer) [reply app/helpers/blacklight/facets_helper_behavior.rb]: 🤷‍♂️ , sure.

File: app/presenters/blacklight/document_presenter.rb [hunk 2]
```diff
@@ -93,7 +93,8 @@ def retrieve_values(field_config)
     deprecation_deprecate retrieve_values: 'Use FieldPresenter#values'
 
     def field_presenter(field_config, options = {})
-      FieldPresenter.new(view_context, document, field_config, options)
+      presenter_class = field_config.presenter || Blacklight::FieldPresenter
```
Comment (jcoyne) [inline app/presenters/blacklight/document_presenter.rb:96]: That looks great. Can the same be done here?

Comment (cbeer) [reply app/presenters/blacklight/document_presenter.rb:96]: Not really. We don't have a class for just index/show fields.

**Pull request merged**